src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -492,14 +492,15 @@ trait TypeDiagnostics {
           && !sym.isParamAccessor       // could improve this, but it's a pain
           && !sym.isEarlyInitialized    // lots of false positives in the way these are encoded
           && !(sym.isGetter && sym.accessed.isEarlyInitialized)
+          && (settings.warnMacros || !sym.isMacro)
         )
         def qualifiesType(sym: Symbol) = !sym.isDefinedInPackage
         def qualifies(sym: Symbol) = (
              (sym ne null)
           && (sym.isTerm && qualifiesTerm(sym) || sym.isType && qualifiesType(sym))
         )
 
-        override def traverse(t: Tree): Unit = {
+        override def traverse(t: Tree): Unit = if (settings.warnMacros || !hasMacroExpansionAttachment(t)) {
```

Comment:
Does this mean: lint if `t` has nothing to do with a macro, or if they want linted macros?